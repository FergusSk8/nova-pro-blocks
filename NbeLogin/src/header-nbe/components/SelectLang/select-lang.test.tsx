import React from 'react';
import { shallow } from 'enzyme';

import SelectLang from './index';

jest.mock('umi', () => {
  return {
    __esModule: true,
    getLocale: jest.fn(),
    setLocale: jest.fn(),
  };
});

describe('HeaderNbe', () => {
  const wrapper = shallow(<SelectLang />);

  it('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set the lang', () => {
    const dropdown = wrapper.find('HeaderDropdown').simulate('mouseover');
    const lang = wrapper.find('Menu').exists;
    expect(dropdown).toBeTruthy();
    expect(lang).toBeTruthy();
  });
});
