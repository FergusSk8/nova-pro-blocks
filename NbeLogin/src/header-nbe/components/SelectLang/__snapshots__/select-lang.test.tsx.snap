// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNbe should show correctly 1`] = `
<HeaderDropdown
  overlay={
    <Menu
      onClick={[Function]}
      selectedKeys={
        Array [
          undefined,
        ]
      }
    >
      <MenuItem>
        Espa√±ol
      </MenuItem>
      <MenuItem>
        English
      </MenuItem>
    </Menu>
  }
  placement="bottomRight"
>
  <span
    className=""
  >
    <ForwardRef(GlobalOutlined)
      title="Language"
    />
  </span>
</HeaderDropdown>
`;
